syntax = "proto3";

package HeadProtocol;

// KVStore service
service KVStore
{
	rpc Put(PutRequest) returns (PutReply) {}
	rpc Get(GetRequest) returns (GetReply) {}
}

// The request message containing the key-value pair to put.
message PutRequest
{
	string key = 1;
	string value = 2;
}

// The response message containing the result of the put operation.
message PutReply { bool success = 1; }

// The request message containing the key to get the value for.
message GetRequest { string key = 1; }

// The response message containing the value for the requested key.
message GetReply { string value = 1; }

// Array transfer service
service ArrayTransfer { rpc SendArray(FloatArray) returns (FloatArray); }
// service ArrayTransfer { rpc SendArray(IntArray) returns (IntArray); }

message IntArray { repeated int32 values = 1; }
message FloatArray { repeated float values = 1; }
